Tests/AssignToManyTests.swift

```diff
@@ -64,6 +67,77 @@ class AssignToManyTests: XCTestCase {
         XCTAssertEqual(dest2.ivar, "Meh")
         XCTAssertEqual(dest3.value, "MehMehMeh")
     }
+
+    func testWeakRetain() {
+        let completion: RetainCheckCompletion = {
+            $0.forEach {
+                XCTAssertEqual($0.initialRetainCount, $0.resultRetainCount)
+            }
+        }
+        checkAssign2RetainValue(for: .weak, completion: completion)
+        checkAssign3RetainValue(for: .weak, completion: completion)
+    }
+
+    func testUnownedRetain() {
+        let completion: RetainCheckCompletion = {
+            $0.forEach {
+                XCTAssertEqual($0.initialRetainCount, $0.resultRetainCount)
+            }
+        }
+        checkAssign2RetainValue(for: .unowned, completion: completion)
+        checkAssign3RetainValue(for: .unowned, completion: completion)
+    }
+
+    func testStrongRetain() {
+        let completion: RetainCheckCompletion = {
+            $0.forEach {
+                XCTAssertEqual($0.initialRetainCount + 1, $0.resultRetainCount)
+            }
+        }
+        checkAssign2RetainValue(for: .strong, completion: completion)
+        checkAssign3RetainValue(for: .strong, completion: completion)
+    }
+
+    private func checkAssign2RetainValue(for ownership: ObjectOwnership, completion: RetainCheckCompletion) {
+        let source = PassthroughSubject<Int, Never>()
+        let dest1 = Fake1(prop: 0)
+        let dest2 = Fake2(ivar: 0)
+        let initialRetainCount1 = CFGetRetainCount(dest1)
+        let initialRetainCount2 = CFGetRetainCount(dest2)
+
+        subscription = source
+            .assign(to: \.prop, on: dest1, and: \.ivar, on: dest2, ownership: ownership)
+
+        let resultRetainCount1 = CFGetRetainCount(dest1)
+        let resultRetainCount2 = CFGetRetainCount(dest2)
+
+        completion([(initialRetainCount1, resultRetainCount1),
```

Comment:
Your tests don't actually seem to check the falues of dest1 and dest2 changes, right? 
Also seems like you're not pushing any value into `source` which might be the reason it's not covered